### ISSUE TITLE
Release Java 8 version using native Java 8 Time Library

### ISSUE THREAD

**Issue by SvenssonWeb** [jmrozanec/cron-utils#254]
Utilizing this in Java 8 requires either, that the project uses Guava and ThreeTenBP replacements in the entire code base, or significant boilerplate to map between Guava/ThreeTenBP and Java 8.

Would a separate release branch for Java 8 be feasible?

I could drive with the migration in that case.

**Comment by jmrozanec** [comment]
@SvenssonWeb Hi! A separate branch for JDK8 is feasible. If we do so, can we add a classifier to the pom.xml, so that we can provide multiple jars for a single version?
You are most welcome to migrate to JDK8. We can publish a branch for this tonight. Thanks! 😄

**Comment by SvenssonWeb** [comment]
@jmrozanec Absolutely, that shouldn't be an issue I think. 
Should I add a branch or did you mean that you were going to?

**Comment by meincs** [comment]
@SvenssonWeb More or less this issue duplicates #175. However, I greatly appreciate any efforts and am looking forward to the result.

**Comment by SvenssonWeb** [comment]
Very much so, I did not look for duplicates of this issue because of the rationale of back-porting to android.
Did you have any conclusive decisions regarding this though?
Removing the dependency to Guava ToStringBuilder would remove Guava as a dependency entirely as well.

I'd love to do the migration, just point me to where you want the change ;)

**Comment by jmrozanec** [comment]
@SvenssonWeb @meincs the original strategy was to wait until Android provides support for JDK8. Maybe this may take too long, so we are OK with maintaining two versions: one with full JDK8 support and another one for Android. Moving to JDK provides multiple advantages: we get rid of Guava and we use the new JDK Time API - both decisions were made to make [cron-utils](https://github.com/jmrozanec/cron-utils) available to a broader set of developers. 

The branch for JDK8 was created and is available [here](https://github.com/jmrozanec/cron-utils/tree/jdk8)

**Comment by amanteaux** [comment]
FYI Android O support the Java 8 time API: <https://developer.android.com/reference/java/time/package-summary.html>

**Comment by meincs** [comment]
@SvenssonWeb Feel free to entirely drop the Guava dependency. For debugging purposes it would be helpful to replace rather drop the toString implementations. You may have a look at https://stackoverflow.com/questions/32691387/alternative-to-moreobjects-in-java-8. Personally I would prefer alternative 3 over the checked 16.

@jmrozanec Hope that efforts on this issue do not cause a further delay of the next 6.0.X release. Although it would be charming to directly switch over to the new Java8 branch.

**Comment by SvenssonWeb** [comment]
@meincs When suggesting "dropping", I meant replacing the dependency with a native solution, not removing the toString() implementation entirely. But I was unclear in this.
And by alt. 3 you mean Tagir Valeev's answer? 

@jmrozanec I'll start building a pull request immediately.

**Comment by meincs** [comment]
@SvenssonWeb yep, by alt. 3 I mean Tagir Valeev's answer.

**Comment by jmrozanec** [comment]
@amanteaux thank you for the update! 
@SvenssonWeb Thanks! We will be glad to merge back to 'master'
@meincs we will release a new version, and later a new one with the migration to JDK8.

**Comment by SvenssonWeb** [comment]
@jmrozanec Just one quick question.
Do you guys have a style guide of any sort?
I saw #111 but didn't really pick up on any direct answer.

The reason I'm asking is because I find a lot of expressions that can be exchanged for lambdas, as well as minor discrepancies regarding newlines and spaces around { and }.

**Comment by SvenssonWeb** [comment]
Seams like Codacy is not configured to review branch jdk8.

**Comment by Naxos84** [comment]
You could use checkstyle plugin for formatting Code.

Am 09.11.2017 17:11 schrieb "jmrozanec" <notifications@github.com>:

**Comment by jmrozanec** [comment]
@SvenssonWeb Codacy is not configured for that branch. Will check locally asap and merge! Thanks!

**Comment by SvenssonWeb** [comment]
@jmrozanec @Naxos84 For us using IntelliJ over Eclipse i could include the codeStyleSettings.xml for code styles in JetBrains IDEs

**Comment by jmrozanec** [comment]
@SvenssonWeb I am ok with it.

**Comment by jmrozanec** [comment]
@SvenssonWeb @meincs @amanteaux @Naxos84 we merged the PR to master! Thank you for helping on this! @SvenssonWeb thank you for migrating to JDK8 and @meincs @amanteaux and @Naxos84 for your comments and suggestions!
@curioustechizen we released a last version with JDK7 support from master, including all features and bugfixes up to now, so that can be leveraged by older Android versions. From now on we will focus on JDK8.
